app/controllers/bookmarks_controller.rb

```diff
@@ -196,4 +200,30 @@ def move_action documents
       BulkActionJobs::Move.perform_later success_ids, params
     end
   end
+
+  def intercom_push_action documents
+    errors = []
+    success_ids = []
+    intercom = Avalon::Intercom.new(current_user.user_key)
+    collections = intercom.user_collections(true)
+    session[:intercom_collections] = collections
+    if intercom.collection_valid?(params[:collection_id])
+      Array(documents.map(&:id)).each do |id|
+        media_object = MediaObject.find(id)
+        if cannot? :intercom_push, media_object
+          errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+        else
+          success_ids << id
+        end
+      end
+      if success_ids.present?
+        session[:intercom_default_collection] = params[:collection_id]
+        BulkActionJobs::IntercomPush.perform_later success_ids, current_user.user_key, params
```

Comment:
This might be a bit paranoid, but I wonder if we should do `params.slice(:collection_id, :include_structure)` to avoid potential serialization/deserialization hacks since `params` would include anything the user submitted.

Reply:
@bkeese and I discussed this and decided that a follow on ticket can be created to apply the slice to each of the bulk actions in bookmarks_controller if it is determined this is a real concern.